lightning/src/ln/msgs.rs

```diff
@@ -610,7 +611,7 @@ pub(crate) struct OnionRealm0HopData {
 	pub(crate) short_channel_id: u64,
 	pub(crate) amt_to_forward: u64,
 	pub(crate) outgoing_cltv_value: u32,
-	// 12 bytes of 0-padding
+	pub(crate) padding: PhantomData<[u8; 12]>,
```

Comment:
It's too bad RFC 2000 (https://github.com/rust-lang/rust/issues/44580) isn't in the stable Rust release yet. Then we could type alias this as:

```rust
pub(crate) type BytePadding<const N: usize> = ::std::marker::PhantomData<[u8; N]>;
```

So this should be fine for now.

Reply:
We should probably do something like this anyway without generic constants. PhantomData has a somewhat specific meaning in Rust (that we have a template type that we're not going to instantiate). We probably just want a new type in ser.rs like (but with better names):
```

pub(crate) struct PaddingData<T: Default + std::convert::AsRef<[u8]>> {
t: std::marker::PhantomData<T>,
}
impl<T: Default + std::convert::AsRef<[u8]>> Writeable for PaddingData<T> {
	#[inline]
	fn write<W: Writer>(&self, w: &mut W) -> Result<(), ::std::io::Error> {
        let nt: T = Default::default();
        w.write_all(nt.as_ref())
	}
}
```